avocado/core/job.py

```diff
@@ -564,7 +570,6 @@ def run(self):
             self.exitcode |= exit_codes.AVOCADO_FAIL
             return self.exitcode
         finally:
-            self.job_pre_post_dispatcher.map_methods('post', self)
```

Comment:
I believe all post_job methods should be called in this `finally` block. I'm talking about result_dispatcher, post_job and archiver as the last thing.